### PR TITLE
dev-python/pylibmc: add py3.5, doc

### PR DESCRIPTION
**Pull request by sbraz**
@SoapGentoo 
```diff
--- pylibmc-1.5.1.ebuild	2017-01-19 13:08:25.352938373 +0100
+++ pylibmc-1.5.1-r1.ebuild	2017-01-19 13:08:54.203089194 +0100
@@ -1,30 +1,47 @@
-# Copyright 1999-2016 Gentoo Foundation
+# Copyright 1999-2017 Gentoo Foundation
 # Distributed under the terms of the GNU General Public License v2
 # $Id$
 
-EAPI="6"
-PYTHON_COMPAT=( python{2_7,3_4} pypy )
+EAPI=6
+PYTHON_COMPAT=( python{2_7,3_4,3_5} )
 
 inherit distutils-r1
 
 DESCRIPTION="Libmemcached wrapper written as a Python extension"
 HOMEPAGE="http://sendapatch.se/projects/pylibmc/ https://pypi.python.org/pypi/pylibmc"
-SRC_URI="mirror://pypi/${PN:0:1}/${PN}/${P}.tar.gz"
+# One image is missing from the doc at PyPI
+# https://github.com/lericson/pylibmc/pull/221
+SRC_URI="https://github.com/lericson/${PN}/archive/${PV}.tar.gz -> ${P}-github.tar.gz"
 
 LICENSE="BSD"
 SLOT="0"
 KEYWORDS="~amd64 ~x86"
-IUSE="test"
+IUSE="doc test"
 
 RDEPEND=">=dev-libs/libmemcached-0.32"
+# Older sphinx versions fail to compile the doc
+# https://github.com/sphinx-doc/sphinx/issues/3266
 DEPEND="${RDEPEND}
-	test? ( dev-python/nose[${PYTHON_USEDEP}] )"
+	doc? ( >=dev-python/sphinx-1.5.1[${PYTHON_USEDEP}] )
+	test? (
+		net-misc/memcached
+		dev-python/nose[${PYTHON_USEDEP}]
+	)"
 
 python_prepare_all() {
-	sed -e "/with-info=1/d" -i setup.cfg
+	sed -e "/with-info=1/d" -i setup.cfg || die
 	distutils-r1_python_prepare_all
 }
 
+python_compile_all() {
+	if use doc; then
+		# This variable is added to sys.path
+		# but it does not seem to be useful
+		PYLIBMC_DIR="." emake -C docs
+		HTML_DOCS=( docs/_build/html/. )
+	fi
+}
+
 python_test() {
 	memcached -d -p 11219 -u nobody -l localhost -P "${T}/m.pid" || die
 	MEMCACHED_PORT=11219 nosetests

```

### CODE REVIEW

**Comment by jbergstroem** [comment]
since sphinx only is used for docs is there a way to retain pypy support? Would be a shame to drop it for doc generation only.

**Comment by sbraz** [comment]
I discussed that with Soap and I'll try to make sphinx 1.5.1 work with pypy. I guess we can put this PR on hold.

**Comment by jbergstroem** [comment]
@sbraz: perhaps its possible to call another interpreter (pypy can't be the only installed python, right?) for documentation generation?

File: dev-python/pylibmc/pylibmc-1.5.1-r1.ebuild [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+LICENSE="BSD"
+SLOT="0"
+KEYWORDS="~amd64 ~x86"
+IUSE="doc test"
+
+RDEPEND=">=dev-libs/libmemcached-0.32"
+# Older sphinx versions fail to compile the doc
+# https://github.com/sphinx-doc/sphinx/issues/3266
+DEPEND="${RDEPEND}
+	doc? ( >=dev-python/sphinx-1.5.1[${PYTHON_USEDEP}] )
```
Comment (SoapGentoo) [inline dev-python/pylibmc/pylibmc-1.5.1-r1.ebuild]: make this
`doc? ( >=dev-python/sphinx-1.5.1[$(python_gen_usedep 'python*')] )`

File: dev-python/pylibmc/pylibmc-1.5.1-r1.ebuild [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+	test? (
+		net-misc/memcached
+		dev-python/nose[${PYTHON_USEDEP}]
+	)"
+
```
Comment (SoapGentoo) [inline dev-python/pylibmc/pylibmc-1.5.1-r1.ebuild]: then add
```
pkg_setup() {
	use doc && DISTUTILS_ALL_SUBPHASE_IMPLS=( 'python*' )
}
```
and finally readd `pypy` (and if you're ultra adventurous, try adding `pypy3`)

File: dev-python/pylibmc/pylibmc-1.5.1-r1.ebuild [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+RDEPEND=">=dev-libs/libmemcached-0.32"
+# Older sphinx versions fail to compile the doc
+# https://github.com/sphinx-doc/sphinx/issues/3266
+DEPEND="${RDEPEND}
+	doc? ( >=dev-python/sphinx-1.5.1[${PYTHON_USEDEP}] )
+	test? (
+		net-misc/memcached
+		dev-python/nose[${PYTHON_USEDEP}]
+	)"
```
Comment (SoapGentoo) [inline dev-python/pylibmc/pylibmc-1.5.1-r1.ebuild]: then add
`REQUIRED_USE="doc? ( || ( $(python_gen_useflags 'python*' ) ) )"`

**Comment by sbraz** [comment]
@jbergstroem Tests don't even pass with pypy :/
```
Traceback (most recent call last):
  File "/var/tmp/portage/dev-python/pylibmc-1.5.1-r1/work/pylibmc-1.5.1/tests/test_refcounts.py", line 97, in test_incr
    initial_refcounts = get_refcounts(refcountables)
  File "/var/tmp/portage/dev-python/pylibmc-1.5.1-r1/work/pylibmc-1.5.1/tests/__init__.py", line 43, in get_refcounts
    return [sys.getrefcount(val) for val in refcountables]
AttributeError: 'module' object has no attribute 'getrefcount'
```
I updated the commit message but I'm afraid we have to get rid of it anyway.

**Pull request closed without merging**